File: Src/Icm.ContextConsole/ApplicationFactory.vb
Imports Icm.Localization
Imports System.Resources
Imports Icm.Tree

''' <summary>
''' Various methods for building and running an Icm.ContextConsole application.
''' </summary>
''' <remarks></remarks>
Public Module ApplicationStarter

    Public Sub Start(Of TMain As IContext)(locRepo As ILocalizationRepository)
        StandardApplication.Create(Of TMain)(extLocRepo:=locRepo).Run()
    End Sub

    Public Sub Start(Of TMain As IContext)(resourceManager As ResourceManager)
        Start(Of TMain)(resourceManager.ToRepository)
    End Sub

    Public Sub Start(rootContextType As Type, resourceManager As ResourceManager)
        Dim app = New StandardApplication(rootContextType, extLocRepo:=resourceManager.ToRepository)
        app.Run()
    End Sub

End Module

File: Src/Icm.ContextConsole/ApplicationFactory.cs
using System;
using Icm.Localization;
using System.Resources;

/// <summary>
/// Various methods for building and running an Icm.ContextConsole application.
/// </summary>
/// <remarks></remarks>
public static class ApplicationStarter
{

	public static void Start<TMain>(ILocalizationRepository locRepo) where TMain : IContext
	{
		StandardApplication.Create<TMain>(extLocRepo: locRepo).Run();
	}

	public static void Start<TMain>(ResourceManager resourceManager) where TMain : IContext
	{
		Start<TMain>(resourceManager.ToRepository());
	}

	public static void Start(Type rootContextType, ResourceManager resourceManager)
	{
		var app = new StandardApplication(rootContextType, extLocRepo: resourceManager.ToRepository());
		app.Run();
	}

}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
